user_guide_src/source/concepts/factories.rst

```diff
@@ -9,32 +9,84 @@ Factories
 Introduction
 ============
 
-Like ``Services``, ``Factories`` are an extension of autoloading that helps keep your code
-concise yet optimal, without having to pass around object instances between classes. At its
+What is Factories?
+------------------
+
+Like :doc:`./services`, **Factories** are an extension of autoloading that helps keep your code
+concise yet optimal, without having to pass around object instances between classes.
+
+At its
 simplest, Factories provide a common way to create a class instance and access it from
 anywhere. This is a great way to reuse object states and reduce memory load from keeping
 multiple instances loaded across your app.
 
-Anything can be loaded by Factories, but the best examples are those classes that are used
+Any class can be loaded by Factories, but the best examples are those classes that are used
 to work on or transmit common data. The framework itself uses Factories internally, e.g., to
 make sure the correct configuration is loaded when using the ``Config`` class.
 
-Take a look at ``Models`` as an example. You can access the Factory specific to ``Models``
-by using the magic static method of the Factories class, ``Factories::models()``. Because of
-the common path structure for namespaces and folders, Factories know that the model files
-and classes are found within **Models**, so you can request a model by its shorthand base name:
+Differences from Services
+-------------------------
+
+Factories require a concrete class name to instantiate, and do not have code to create instances.
+
+So Factories are not good for creating complex instance that needs many dependencies,
+and cannot change the class of the instance to be returned.
+
+On the other hand, Services have code to create instances, so it can create a complex instance
+that needs other services or class instances. When you get a service, Services require a service name,
+not a class name, so the returned instance can be changed without changing the client code.
```

Comment:
```suggestion
Factories require a concrete class name to instantiate and do not have code to create instances.

So, Factories are not good for creating a complex instance that needs many dependencies,
and you cannot change the class of the instance to be returned.

On the other hand, Services have code to create instances, so it can create a complex instance
that needs other services or class instances. When you get a service, Services require a service name,
not a class name, so the returned instance can be changed without changing the client code.
```